hack/zstd-dict/main.go

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) 2024 Sidero Labs, Inc.
```

Comment:
I would be nice to have some README on how this intended to be ran and where.
I Kubernetes? Or locally?

I guess the goal is to generate `config.zdict` right?

Reply:
Yes, I'll write it down to clarify. This code does two things
- as you said, generate a config.zdict. But then you are responsible of putting it to its place.
- it also writes all the input files of the dictionary to the `files/` dir there, so `zstd` command line tool can be used to generate the dictionary instead (gives better result than the go code).

I'll write them down nicely in README.

Reply:
done

Reply:
Also removed the argo manifests - now the dict tool downloads the chart and does `helm template` on it itself.